app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/adapter/LiveDataListViewAdapter.java

```diff
@@ -0,0 +1,322 @@
+        mItems.set(position, item);
+        notifyItemRangeChanged(position, 1);
+    }
+
+    /**
+     * Removes a range of items from the adapter. The range is specified by giving
+     * the starting position and the number of elements to remove.
+     *
+     * @param position The index of the first item to remove.
+     * @param count    The number of items to remove.
+     * @return The number of items removed.
+     */
+    public int removeItems(int position, int count) {
+        int itemsToRemove = Math.min(count, mItems.size() - position);
+        if (itemsToRemove <= 0) {
+            return 0;
+        }
+
+        for (int i = 0; i < itemsToRemove; i++) {
+            mItems.remove(position);
+        }
+        notifyItemRangeRemoved(position, itemsToRemove);
+        return itemsToRemove;
+    }
+
+    /**
+     * Removes all items from this adapter, leaving it empty.
+     */
+    public void clear() {
+        int itemCount = mItems.size();
+        if (itemCount == 0) {
+            return;
+        }
+        mItems.clear();
+        notifyItemRangeRemoved(0, itemCount);
+    }
+
+    /**
+     * Gets a read-only view of the list of object of this ArrayObjectAdapter.
+     */
+    public <E> List<E> unmodifiableList() {
+        return Collections.unmodifiableList((List<E>) mItems);
+    }
+
+    @Override
+    public boolean isImmediateNotifySupported() {
+        return true;
+    }
+
+    /**
+     * Set a new item list to adapter. The Diffutil will compute the difference and dispatch it to
+     * according position
+     *
+     * @param itemList              List of new Items
+     * @param sameItemComparator    The comparator to determine if two item are same or not
+     * @param sameContentComparator The comparator to determin if two item's content are same or not
+     */
+    public void setItems(final List<T> itemList, final Comparator<T> sameItemComparator,
+                         final Comparator<T> sameContentComparator) {
+
+        DiffUtil.DiffResult result = DiffUtil.calculateDiff(new DiffUtil.Callback() {
+            @Override
+            public int getOldListSize() {
+                return mItems.size();
+            }
+
+            @Override
+            public int getNewListSize() {
+                return itemList.size();
+            }
+
+            @Override
+            public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
+                return sameItemComparator.compare(mItems.get(oldItemPosition),
+                        itemList.get(newItemPosition)) == 0;
+            }
+
+            @Override
+            public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
+                return sameContentComparator.compare(mItems.get(oldItemPosition),
+                        itemList.get(newItemPosition)) == 0;
+            }
+        });
+        mItems = itemList;
+        result.dispatchUpdatesTo(recyclerViewAdapter);
```

Comment:
dont use a fake RecyclerView Adapter to get the observer, write own callback and fire leanback ObjectAdapter event.

Reply:
Done